internal/syncer/gosec_repo_scan.go

```diff
@@ -0,0 +1,147 @@
+package syncer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/mergestat/mergestat/internal/db"
+	"github.com/mergestat/mergestat/internal/helper"
+)
+
+// gosecIssue represents an issue identified by gosec.
+// The struct is documented here: https://pkg.go.dev/github.com/securego/gosec/v2#Issue
+type gosecIssue struct {
+	Severity   string `json:"severity"`   // issue severity (how problematic it is)
+	Confidence string `json:"confidence"` // issue confidence (how sure we are we found it)
+	Cwe        struct {
+		ID          string
+		Name        string
+		Description string
+	} `json:"cwe"` // Cwe associated with RuleID
+	RuleID string `json:"rule_id"` // Human-readable explanation
+	What   string `json:"details"` // Human-readable explanation
+	File   string `json:"file"`    // File name we found it in
+	Code   string `json:"code"`    // Impacted code line
+	Line   string `json:"line"`    // Line number in file
+	Col    string `json:"column"`  // Column number in line
+	NoSec  bool   `json:"nosec"`   // true if the issue is nosec
+}
+
+// handleGosecRepoScan executes `gosec -q -fmt json {git-repo}` for a repo
+// and inserts the output JSON into the DB
+func (w *worker) handleGosecRepoScan(ctx context.Context, j *db.DequeueSyncJobRow) error {
+	l := w.loggerForJob(j)
+
+	tmpPath, cleanup, err := helper.CreateTempDir(os.Getenv("GIT_CLONE_PATH"), "mergestat-repo-")
+	if err != nil {
+		return fmt.Errorf("temp dir: %w", err)
+	}
+	defer func() {
+		if err := cleanup(); err != nil {
+			l.Err(err).Msgf("error cleaning up repo at: %s, %v", tmpPath, err)
+		}
+	}()
+
+	var ghToken string
+	if ghToken, err = w.fetchGitHubTokenFromDB(ctx); err != nil {
+		return err
+	}
+
+	if err = w.cloneRepo(ctx, ghToken, j.Repo, tmpPath, false, j); err != nil {
+		return fmt.Errorf("git clone: %w", err)
+	}
+
+	// indicate that we're starting a gitleaks scan
+	if err := w.sendBatchLogMessages(ctx, []*syncLog{{Type: SyncLogTypeInfo, RepoSyncQueueID: j.ID,
+		Message: fmt.Sprintf(LogFormatStartingSync, j.SyncType, j.Repo),
+	}}); err != nil {
+		return fmt.Errorf("send batch log messages: %w", err)
+	}
+
+	var stdout, stderr bytes.Buffer
+	cmd := exec.CommandContext(ctx, "gosec", "-no-fail", "-fmt", "json", "./...")
```

Comment:
```suggestion
	cmd := exec.CommandContext(ctx, "gosec", "-no-fail", "-fmt", "json", "./...")
```

curious about the `"./..."` arg here - shouldn't it be the root of the cloned repo?

Reply:
so, if I run `gosec` w/o any path arguments, it errors saying:

```shell
> gosec -no-fail -fmt json
Error: FILE [FILE...] or './...' or -r expected

# ... usage
```

I tried the `-r` flag but it seems to be not implemented 🙄 So, instead, I added `./...` and set the `cmd.Dir = tmpPath` (the root of the cloned repository)

Reply:
we could've done `$tmpPath/...` but I'm not sure how `gosec` would behave (does it treat it as an import?)